Deleting all legacy PACKAGE.py

souravpy opened the pull request:
Deleted following files as remarked
```
./GangaCore/PACKAGE.py
./GangaDirac/PACKAGE.py
./GangaGaudi/PACKAGE.py
./GangaLHCb/PACKAGE.py
./GangaND280/PACKAGE.py
./GangaTest/Lib/GListApp/PACKAGE.py
./GangaTest/Lib/TFile/PACKAGE.py
./GangaTest/PACKAGE.py
./GangaTutorial/PACKAGE.py
```

souravpy committed:
Delete ganga/GangaDirac/PACKAGE.py

souravpy committed:
Delete ganga/GangaGaudi/PACKAGE.py

souravpy committed:
Delete ganga/GangaLHCb/PACKAGE.py

souravpy committed:
Delete ganga/GangaND280/PACKAGE.py

souravpy committed:
Delete ganga/GangaTest/Lib/GListApp/PACKAGE.py

souravpy committed:
Delete ganga/GangaTest/Lib/TFile/PACKAGE.py

souravpy committed:
Delete ganga/GangaTest/PACKAGE.py

souravpy committed:
Delete ganga/GangaTutorial/PACKAGE.py

souravpy committed:
Delete ganga/GangaCore/PACKAGE.py

souravpy committed:
Merge branch 'develop' into develop

egede commented:
Some tests are failing, reporting a failure of `import GangaCore.PACKAGE`. It should be investigated if those lines can just be deleted.

egede commented:
The `PackageSetup` class should be deleted as well.

souravpy committed:
Update Setup.py

Removed PackageSetup class

souravpy committed:
Merge branch 'develop' into develop

souravpy commented:
> The `PackageSetup` class should be deleted as well.

Done as suggested, kindly check the PR

souravpy committed:
Update fill_gpi.py

Removed GangaCore PACKAGE.py dependency

alexanderrichards commented on ganga/GangaCore/Utility/Setup.py:
```diff
@@ -1,141 +1,5 @@
-
-        return True
-
+class PackageSet
```
should this be:

```python
class PackageSetup:
```

alexanderrichards replied:
I missed what @egede said above, looks like this can just go

souravpy replied:
Okayy

souravpy committed:
Update GangaUnitTest.py

Removed GangaCore PACKAGE.py dependency

souravpy committed:
Update ganga

Removed GangaCore PACKAGE.py dependency

souravpy committed:
Update TestStartUp.py

Removed GangaCore PACKAGE.py dependency

souravpy committed:
Update Setup.py

suggested edits

egede commented:
So there are still some failing tests. I think that they are still casued by lack of cleaning up. If I search for `PACKAGE` in the code base, I still find quite a few loose ends. The files identified below should all be investigated and the lines potentially deleted.
```
(ganga) [desktop] ~/.../ganga/ganga/ganga souravpy/develop % grep -r PACKAGE 2> /dev/null | grep -v '^[A-Za-z_/.0-9]*:\s*[#"]' | grep -v APP_PACKAGE
GangaCore/Utility/Runtime.py:            __import__(self.name + ".PACKAGE")
GangaCore/Utility/Setup.py:    if GangaCore.PACKAGE._defaultPlatform != platform:
GangaCore/Utility/Setup.py:                    % (GangaCore.PACKAGE._defaultPlatform, platform))
GangaCore/Utility/Setup.py:    GangaCore.PACKAGE._defaultPlatform = _new_platform = platform
GangaCore/Utility/Setup.py:    from GangaCore.PACKAGE import _defaultPlatform
GangaCore/Utility/Setup.py:    GangaCore.PACKAGE._defaultExternalHome = externalHome
GangaCore/Utility/Setup.py:    from GangaCore.PACKAGE import _defaultExternalHome
GangaDirac/__init__.py:    from . import PACKAGE
GangaDirac/__init__.py:    PACKAGE.standardSetup()
GangaGaudi/__init__.py:    from . import PACKAGE
GangaGaudi/__init__.py:    PACKAGE.standardSetup()
GangaLHCb/__init__.py:    from . import PACKAGE
GangaLHCb/__init__.py:    PACKAGE.standardSetup()
GangaND280/BOOT.py:    from . import PACKAGE
GangaND280/BOOT.py:    PACKAGE.standardSetup()
GangaTest/Framework/runner.py:myConfig.addOption('SearchReleaseTests', True, 'search for tests packaged in release (PACKAGE/old_test dir) lookup')
GangaTest/__init__.py:    import GangaTest.PACKAGE
GangaTest/__init__.py:    PACKAGE.standardSetup()
GangaTutorial/BOOT.py:    import PACKAGE
GangaTutorial/BOOT.py:    PACKAGE.standardSetup()
```

Remember that you can run the tests locally with 

```
python -m pytest ganga/GangaCore/test/Unit
```
as long as you have installed the `dev` extension package, i.e. in the directory containing the `setup.py` file of the cloned ganga directory, you install dependencies with 
```
pip install .[dev]
```

souravpy commented:
Yes, now I got that keyword here is test everything before pushing code, back to resolving the issues.

mesmith75 committed:
Merge branch 'develop' into develop

souravpy committed:
Update Runtime.py

resolving:
GangaCore/Utility/Runtime.py:            __import__(self.name + ".PACKAGE")

souravpy committed:
Update Setup.py

souravpy committed:
Update __init__.py

souravpy committed:
Update __init__.py

souravpy committed:
Update __init__.py

souravpy committed:
Update BOOT.py

souravpy committed:
Update runner.py

souravpy committed:
Update __init__.py

souravpy committed:
Update BOOT.py

souravpy committed:
Update __init__.py

Resolving:
ERROR ganga/GangaCore/test/Condor/TestCondor.py::test_job_a_create -   File "/__w/ganga/ganga/ganga/GangaTest/__init__.py", line 7
    def loadPlugins(config=None):
    ^^^
IndentationError: expected an indented block after function definition on line 2

souravpy committed:
Merge branch 'develop' into develop

mesmith75 commented on ganga/GangaCore/Utility/Runtime.py:
```diff
@@ -154,7 +154,7 @@ def __init__(self, path):
             # in this way we enforce any initialization of module is performed
             # (e.g PackageSetup.setPlatform() is called)
-            __import__(self.name + ".PACKAGE")
+            #__import__(self.name + ".PACKAGE")
```
please delete rather than comment

mesmith75 commented on ganga/GangaDirac/__init__.py:
```diff
@@ -156,10 +156,10 @@
 
 
-def standardSetup():
+#def standardSetup():
```
delete

souravpy committed:
Update Runtime.py

souravpy committed:
Update __init__.py

souravpy committed:
Update BatchScriptTemplate.py.template

Remove legacy python2 dependencies #2222

souravpy committed:
Update Notebook.py

souravpy committed:
Merge pull request #1 from souravpy/souravpy-patch-4

Remove legacy python2 dependencies #2222

souravpy committed:
Revert "Remove legacy python2 dependencies #2222"

souravpy committed:
Merge pull request #2 from souravpy/revert-1-souravpy-patch-4

Revert "Remove legacy python2 dependencies #2222"

mesmith75 commented:
Could you please delete the references to package, rather than just comment.

souravpy commented:
> Could you please delete the references to package, rather than just comment.

Okayy

souravpy commented:
I think Code works, and passes all test?

souravpy committed:
Delete ganga/GangaCore/Lib/Notebook/Notebook.py

mesmith75 commented on ganga/GangaCore/Lib/Notebook/Notebook.py:
```diff
@@ -1,199 +0,0 @@
-##########################################################################
```
I am not sure we want to delete this

mesmith75 commented on ganga/GangaGaudi/__init__.py:
```diff
@@ -18,10 +18,10 @@
 
 
-def standardSetup():
+#def standardSetup():
```
These need deleting

mesmith75 commented on ganga/GangaLHCb/__init__.py:
```diff
@@ -94,10 +94,10 @@ def _store_root_version():
 
 
-def standardSetup():
+#def standardSetup():
```
These need deleting

mesmith75 commented on ganga/GangaND280/BOOT.py:
```diff
@@ -1,6 +1,6 @@
 def getEnvironment(c):
-    from . import PACKAGE
-    PACKAGE.standardSetup()
```
These need deleting

mesmith75 commented on ganga/GangaTest/Framework/runner.py:
```diff
@@ -55,7 +55,7 @@
 myConfig.addOption('SearchLocalTests', True, 'search for local tests lookup')
 # search for tests packaged in release (PACKAGE/old_test dir) lookup
-myConfig.addOption('SearchReleaseTests', True, 'search for tests packaged in release (PACKAGE/old_test dir) lookup')
+#myConfig.addOption('SearchReleaseTests', True, 'search for tests packaged in release (PACKAGE/old_test dir) lookup')
```
Why is this commented?

mesmith75 commented on ganga/GangaTutorial/BOOT.py:
```diff
@@ -1,6 +1,6 @@
 def getEnvironment(c):
-    import PACKAGE
-    PACKAGE.standardSetup()
+    #import PACKAGE
```
delete

souravpy committed:
Update __init__.py

souravpy committed:
Update __init__.py

souravpy committed:
Update BOOT.py

souravpy committed:
Update runner.py

souravpy committed:
Update BOOT.py

mesmith75 commented on ganga/GangaCore/Utility/Runtime.py:
```diff
@@ -154,8 +154,7 @@ def __init__(self, path):
             # @see Ganga/PACKAGE.py for description of this magic module
             # in this way we enforce any initialization of module is performed
             # (e.g PackageSetup.setPlatform() is called)
-            __import__(self.name + ".PACKAGE")
```
The comments above relating to this need removing.

mesmith75 commented on ganga/GangaCore/Utility/Setup.py:
```diff
@@ -180,19 +44,15 @@ def setPlatform(platform):
             "Ganga platform has been already set to: %s" % str(_new_platform))
     # change the global platform used in bootstrap
     import Ganga
-    if GangaCore.PACKAGE._defaultPlatform != platform:
```
This function is now broken, as are several of the functions below (`getPlatform`, `setExternalHome`, `getExternalHome`). If the code in this class does nothing then we can get rid of it, otherwise it probably wants fixing.

Does ganga work with these changes?

egede replied:
I am removing the entire file as nothing in it called now.

mesmith75 commented on ganga/GangaTest/Framework/runner.py:
```diff
@@ -55,7 +55,7 @@
 # search for local tests lookup
 myConfig.addOption('SearchLocalTests', True, 'search for local tests lookup')
 # search for tests packaged in release (PACKAGE/old_test dir) lookup
-myConfig.addOption('SearchReleaseTests', True, 'search for tests packaged in release (PACKAGE/old_test dir) lookup')
```
Do we know what this config option does - are we sure it wants to be deleted?

egede replied:
It is never used as it relates to the old testing framework.

souravpy committed:
Update Runtime.py

egede committed:
Merge branch 'develop' into develop

souravpy committed:
Merge branch 'develop' into develop

souravpy committed:
Merge branch 'ganga-devs:develop' into develop

egede committed:
Merge branch 'develop' into develop

egede commented:
We see a set of failures in the testing here, that I assume is a side effect of the code that has been removed.
```
/home/runner/venv/lib/python3.11/site-packages/coverage/inorout.py:503: CoverageWarning: Module ganga/GangaCore/PACKAGE.py was never imported. (module-not-imported)
  self.warn(f"Module {pkg} was never imported.", slug="module-not-imported")
/home/runner/venv/lib/python3.11/site-packages/coverage/inorout.py:503: CoverageWarning: Module ganga/GangaCore/__init__.py was never imported. (module-not-imported)
  self.warn(f"Module {pkg} was never imported.", slug="module-not-imported")
/home/runner/venv/lib/python3.11/site-packages/coverage/report_core.py:109: CoverageWarning: Couldn't parse Python file '/home/runner/work/ganga/ganga/ganga/GangaCore/Utility/Setup.py' (couldnt-parse)
  coverage._warn(msg, slug="couldnt-parse")
collecting ... collected 139 items / 2 errors

==================================== ERRORS ====================================
_____ ERROR collecting ganga/GangaCore/test/Unit/Notebook/TestNotebook.py ______
ImportError while importing test module '/home/runner/work/ganga/ganga/ganga/GangaCore/test/Unit/Notebook/TestNotebook.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ganga/GangaCore/test/Unit/Notebook/TestNotebook.py:15: in <module>
    from GangaCore.Lib.Notebook import Notebook
ganga/GangaCore/Lib/Notebook/__init__.py:2: in <module>
    from .Notebook import Notebook
E   ModuleNotFoundError: No module named 'GangaCore.Lib.Notebook.Notebook'
```
I think the first step is to understand the warinings and the reason for the failure of the import. The nthe fix might actually be easy.

souravpy committed:
Update TestNotebook.py

mesmith75 commented:
I think this has gone in the wrong direction - we want to keep the Notebook, not remove it.

souravpy commented:
> I think this has gone in the wrong direction - we want to keep the Notebook, not remove it.

I'm keeping it, just omitting the line that is giving error on import, I'll fix it asap!

mesmith75 commented:
I'm not sure I understand - it gives you the error because you deleted the file. You need to keep the file and that import.

souravpy committed:
Update TestNotebook.py

egede committed:
Restore deleted file

egede committed:
Merge branch 'develop' into develop

egede commented:
@souravpy I fixed the collision caused by the file that was removed/restored. When developing code on a fork, please use a branch rather than the just the `main` branch (called `develop` here). In the way you have done it here, you have granted all Ganga developers write access to the main branch in your fork. Probably not what you wanted.

egede committed:
Remove unused file

mesmith75 committed:
Merge branch 'develop' into develop
